server/src/main/java/org/elasticsearch/rest/action/search/RestSearchAction.java

```diff
@@ -95,6 +98,8 @@ public Set<String> supportedCapabilities() {
 
     @Override
     public RestChannelConsumer prepareRequest(final RestRequest request, final NodeClient client) throws IOException {
+        // set weather data nodes should send back stack trace based on the `error_trace` query parameter
+        context.putHeader("error_trace", request.param("error_trace", "false"));
```

Comment:
Maybe we could also centralize the putHeader call somewhere. Where do we handle the common rest parameters perhaps?

Reply:
Done!